native-windows-gui/src/win32/clipboard.rs

```diff
@@ -241,31 +241,19 @@ impl Clipboard {
         ptr::copy_nonoverlapping(data, GlobalLock(alloc) as *mut D, count);
         GlobalUnlock(alloc);
 
-        SetClipboardData(fmt, alloc as HANDLE);
+        if SetClipboardData(fmt, alloc as HANDLE).is_null() {
+            GlobalFree(alloc);
+        }
     }
 
     /**
         Check if the selected format is available in the clipboard.
-        The clipboard must be open when calling this function.
     */
     pub fn has_format(fmt: ClipboardFormat) -> bool {
-        use winapi::um::winuser::EnumClipboardFormats;
+        use winapi::um::winuser::IsClipboardFormatAvailable;
 
         let selected_format = fmt.into_raw();
-        let mut format = 0;
-        let next_format = unsafe {
-            |fmt: &mut u32| { let f = EnumClipboardFormats(*fmt); *fmt = f; f }
-        };
-
-        let mut found = false;
-        while next_format(&mut format) != 0 {
-            if format == selected_format {
-                found = true;
-                break;
-            }
-        }
-
-        found
+        unsafe { IsClipboardFormatAvailable(selected_format) != 0 }
```

Comment:
actually now I htink about it, EnumClipboardFormats probably can take into account context(window handle)?

Reply:
Will do some tests on my side just to be sure. Anyway I need to add a clipboard example.